internal/http/client.go

```diff
@@ -62,11 +67,35 @@ func NewClient(config config.Config) NewRelicClient {
 		config.UserAgent = defaultUserAgent
 	}
 
+	if config.LogLevel == "" {
+		config.LogLevel = defaultLogLevel
+	}
+
 	r := retryablehttp.NewClient()
 	r.HTTPClient = &c
 	r.RetryMax = defaultRetryMax
 	r.CheckRetry = RetryPolicy
 
+	if config.Logger != nil {
+		r.Logger = config.Logger
+	} else {
+		r.Logger = logging.StructuredLogger{}
+
+		log.AddHook(&logging.DefaultFieldHook{})
+
+		level, err := log.ParseLevel(config.LogLevel)
+		if err != nil {
+			log.Warn(fmt.Sprintf("could not parse log level '%s', logging will proceed at %s level", config.LogLevel, defaultLogLevel))
+			level, _ = log.ParseLevel(defaultLogLevel)
+		}
+
+		log.SetLevel(level)
+
+		if config.LogJSON {
+			log.SetFormatter(&log.JSONFormatter{})
+		}
+	}
+
```

Comment:
It feels like we should be able to just do `r.Logger = config.Logger` and handle creating a logger within Config.

Reply:
`logging` should handle creation of the default logger.

```
if config.Logger != nil {
  // same
} else {
  r.Logger = logging.DefaultLogger
  // or
  // r.Logger = logging.New(config)
  // or if moving logging into a deeper struct in config:
  // r.Logger = logging.New(config.Logger)
}
```